src/background/logger.js

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import {Component} from './component.js';
+
+let self;
+
+export class Logger extends Component {
+  constructor(receiver) {
+    super(receiver);
+    self = this;
```

Comment:
maybe add `if(!self){}` otherwise creating another instance might create wierd issues.  (if you want to keep it as class)